File: day001/deduplicate.py
def unique_numbers(numbers: list[int]) -> list[int]:
    # 在这里实现，不要直接写死返回结果
    a = []
    for num in numbers:
      if num not in a:
         a.append(num)
    return a


print(unique_numbers([3, 1, 3, 2, 1]))  # 预期：[3, 1, 2]
print(unique_numbers([]))               # 预期：[]
print(unique_numbers([2, 2, 2]))        # 预期：[2]
print(unique_numbers([3, 1, 2]))        # 预期：[3, 1, 2]

File: day001/Deduplicate.java
import java.util.ArrayList;
import java.util.List;

public class Deduplicate {
    public static List<Integer> uniqueNumbers(List<Integer> numbers) {
        List<Integer> result = new ArrayList<>();

        // TODO：遍历 numbers，只把尚未出现的数字加入 result
        for(Integer num:numbers){
            if (!result.contains(num)) {
                result.add(num);
            }
        }
        return result;
    }

    public static void main(String[] args) {
        System.out.println(uniqueNumbers(List.of(3, 1, 3, 2, 1)));
        System.out.println(uniqueNumbers(List.of()));
        System.out.println(uniqueNumbers(List.of(2, 2, 2)));
        System.out.println(uniqueNumbers(List.of(3, 1, 2)));
    }
}